src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,158 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.auth;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.UUID;
+
+import org.jboss.aerogear.android.authorization.AuthzModule;
+import org.jboss.aerogear.android.authorization.oauth2.OAuth2AuthzSession;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.http.HeaderAndBody;
+import org.jboss.aerogear.android.pipe.http.HttpException;
+import org.jboss.aerogear.android.pipe.http.HttpProvider;
+import org.jboss.aerogear.android.pipe.http.HttpRestProvider;
+import org.jboss.aerogear.android.pipe.module.AuthorizationFields;
+import org.jboss.aerogear.android.pipe.module.ModuleFields;
+import org.jboss.aerogear.android.store.DataManager;
+import org.jboss.aerogear.android.store.generator.IdGenerator;
+import org.jboss.aerogear.android.store.sql.SQLStore;
+import org.jboss.aerogear.android.store.sql.SQLStoreConfiguration;
+
+import android.app.Activity;
+import android.content.Context;
+import android.util.Base64;
+import timber.log.Timber;
+
+/**
+ * ModifiedAuthzModule.
+ * <p/>
+ * Performs all related work of generating new key and secret pair, Storing them as account,
+ * Retrieving stored account, Deleting stored account.
+ */
+
+public class SecretStoreAuthzModule implements AuthzModule {
+    private final Context context;
+    private final SQLStore<OAuth2AuthzSession> sessionStore;
+
+    public SecretStoreAuthzModule(Context context) {
+        this.context = context.getApplicationContext();
+        this.sessionStore = openSessionStore();
+        sessionStore.openSync();
+    }
+
+    private SQLStore<OAuth2AuthzSession> openSessionStore() {
+        DataManager.config(AuthData.STORE, SQLStoreConfiguration.class)
+                .forClass(OAuth2AuthzSession.class)
+                .withContext(context)
+                .withIdGenerator(new IdGenerator() {
+                    @Override
+                    public String generate() {
+                        return UUID.randomUUID().toString();
+                    }
+                }).store();
+        return (SQLStore<OAuth2AuthzSession>) DataManager.getStore(AuthData.STORE);
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        return sessionStore.read(AuthData.NAME) != null;
+    }
+
+    public void requestAccess(final Callback<String> callback,
```

Comment:
You shouldn't be doing this.

You should be either a) using an authModule which has support for passing in a UN and PW or b) delegating the colleciton of the UN and PW to an intent and collecting the necessary tokens in a callback similar to how the intent based Oauth2AuthzModule works.